pkg/controllers/clusterresourceplacementeviction/controller_intergration_test.go

```diff
@@ -0,0 +1,1031 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package clusterresourceplacementeviction
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	placementv1alpha1 "go.goms.io/fleet/apis/placement/v1alpha1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+const (
+	crbNameTemplate        = "crb-%d"
+	anotherCRBNameTemplate = "another-crb-%d"
+	crpNameTemplate        = "crp-%d"
+	evictionNameTemplate   = "eviction-%d"
+)
+
+var (
+	lessFuncCondition = func(a, b metav1.Condition) bool {
+		return a.Type < b.Type
+	}
+
+	evictionStatusCmpOptions = cmp.Options{
+		cmpopts.SortSlices(lessFuncCondition),
+		cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),
+		cmpopts.EquateEmpty(),
+	}
+)
+
+const (
+	eventuallyDuration   = time.Minute * 2
+	eventuallyInterval   = time.Millisecond * 250
+	consistentlyDuration = time.Second * 10
+	consistentlyInterval = time.Millisecond * 500
+)
+
+var _ = Describe("Test ClusterResourcePlacementEviction Controller", Ordered, func() {
```

Comment:
We don't need Ordered, ran the IT using,

ginkgo -procs=2 -v -race ./pkg/controllers/clusterresourceplacementeviction/...

To ensure it can run in parallel will move it out of the controller directory and move it test/integration in a follow up PR. Using BeforeEach & AfterEach instead